starfish/image/_stack.py

```diff
@@ -30,15 +30,35 @@ def __init__(self, image_partition):
             self._num_zlayers = 1
         self._tile_shape = tuple(image_partition.default_tile_shape)
 
-        self._data = numpy.zeros((self._num_hybs, self._num_chs, self._num_zlayers) + self._tile_shape)
-        self._data_needs_writeback = False
+        # Examine the tiles to figure out the right kind (int, float, etc.) and size.  We require that all the tiles
```

Comment:
I talked this over with @sofroniewn . 

Because image dtypes are thought of as intensity scales, for example, 0 is minimum intensity and 2 ** n_bits is max intensity, simply merging images with different scales and picking the largest data type won't do what we want it to. 

Instead, we would need to scale the intensity values to whatever dtype is being used. So, if scaling a uint8 image to uint32, you'd need to multiple transform 2 ** 8 --> 2 ** 32. 

I think the largest dtype is a good scaling point (as it is here), although Deep has previously stated that "starfish requests uint16", so a future PR might add a rescaling to that specific dtype here. 

There are tools in skimage to rescale most of the dtypes, see [http://scikit-image.org/docs/dev/user_guide/data_types.html](http://scikit-image.org/docs/dev/user_guide/data_types.html). This also provides some background on why scaling is important. 

If images with multiple dtypes are found, we should emit a warning, as that's definitely weird.